examples/v2/chat2.nim

```diff
@@ -156,16 +154,23 @@ proc printReceivedMessage(c: Chat, msg: WakuMessage) {.raises: [Defect].} =
     trace "Printing message", topic=DefaultTopic, chatLine,
       contentTopic = msg.contentTopic
 
-proc selectRandomNode(fleetStr: string): string =
+proc selectRandomNode(fleetStr: string): Option[string] =
```

Comment:
If a select operation fails, should it not be a Result type that can error?

Alternatively, if we are getting it from a list and the list happens to be empty, then Option makes sense.

Here it seems like the empty case is a HTTP exception, which seems more like an error to me. Semantics though...

Reply:
With error the exception message can be propagated upward too, such as if it is a permission issue, server down, change of format for fleets json, etc.

Reply:
Agree. `Result` seems more idiomatic here.